src/webgl-preview.ts

```diff
@@ -101,21 +107,27 @@ export class WebGLPreview {
   singleLayerMode = false;
   buildVolume?: BuildVolume;
   initialCameraPosition = [-100, 400, 450];
-  debug = false;
-  allowDragNDrop = false;
-  controls: OrbitControls;
-  beyondFirstMove = false;
+  debug = false; // deprecated
+  allowDragNDrop = false; // deprecated
   inches = false;
   nonTravelmoves: string[] = [];
-  _animationFrameId?: number;
   disableGradient = false;
 
-  state: State = { x: 0, y: 0, z: 0, r: 0, e: 0, i: 0, j: 0, t: 0 };
-
-  static readonly defaultExtrusionColor = new Color('hotpink');
+  // gcode processing state
+  private state: State = State.initial;
+  private prevState: State;
+  private prevLayerIndex?: number = undefined;
+  private beyondFirstMove = false;
 
+  // rendering
+  private group?: Group;
   private disposables: { dispose(): void }[] = [];
+  static readonly defaultExtrusionColor = new Color('hotpink');
   private _extrusionColor: Color | Color[] = WebGLPreview.defaultExtrusionColor;
+  private animationFrameId?: number;
+  private renderLayerIndex = 0;
+
+  // colors
```

Comment:
Agree with keeping them all private (except for `defaultExtrusionColor` which can be useful to have access to, and you kept it exposed)